tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js

```diff
@@ -0,0 +1,183 @@
+      });
+
+      this.server.put('/profiles/:id/ds_automated_device_preference', () => {
+        return {
+          ds_automated_device_selection:
+            ENUMS.DS_MANUAL_DEVICE_SELECTION.ANY_DEVICE,
+        };
+      });
+
+      this.server.get(
+        '/profiles/:id/ds_automated_device_preference',
+        (schema) => {
+          console.log(schema);
+          return {
+            ds_automated_device_selection:
+              ENUMS.DS_MANUAL_DEVICE_SELECTION.ANY_DEVICE,
+            ds_automated_platform_version_min: 13,
+          };
+        }
+      );
+
+      this.server.get('/projects/:id/available-devices', (schema) => {
+        const results = schema.projectAvailableDevices.all().models;
+
+        return { count: results.length, next: null, previous: null, results };
+      });
+
+      const normalizedProject = store.normalize('project', {
+        ...project.toJSON(),
+      });
+
+      this.setProperties({
+        project: store.push(normalizedProject),
+      });
+
+      await this.owner.lookup('service:organization').load();
+    });
+
+    test('it renders', async function (assert) {
+      await render(hbs`
+      <ProjectSettings::GeneralSettings::DevicePreferencesAutomatedDast @project={{this.project}}/>
+    `);
+
+      assert
+        .dom(
+          '[data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-title]'
+        )
+        .exists()
+        .containsText('t:devicePreferencesAutomatedDast:()');
+
+      assert
+        .dom(
+          '[data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-preference-select]'
+        )
+        .exists();
+    });
+
+    test('it selects criteria', async function (assert) {
+      await render(hbs`
+      <ProjectSettings::GeneralSettings::DevicePreferencesAutomatedDast @project={{this.project}}/>
+    `);
+
+      let selectListItems = findAll('.ember-power-select-option');
+
+      const anyDeviceLabel = `t:anyDevice:()`;
+
+      // Select "Any Device"
+      await selectChoose(`.${classes.trigger}`, anyDeviceLabel);
+
+      await click(`.${classes.trigger}`);
+      selectListItems = findAll('.ember-power-select-option');
+
+      //  "Any Device" is first option
+      assert.dom(selectListItems[0]).hasAria('selected', 'true');
+
+      assert.dom('[data-test-dast-preference-criteria-table]').doesNotExist();
+
+      //  Select 'Specific Device'
+      const specificDeviceLabel = `t:defineDeviceCriteria:()`;
+
+      await selectChoose(`.${classes.trigger}`, specificDeviceLabel);
+
+      await click(`.${classes.trigger}`);
+
+      //  "Specific Device" is second option
+      selectListItems = findAll('.ember-power-select-option');
+
+      assert.dom(selectListItems[1]).hasAria('selected', 'true');
+
+      await click(`.${classes.trigger}`);
+
+      assert.dom('[data-test-dast-preference-criteria-table]').exists();
+    });
+
+    test('it selects min version', async function (assert) {
```

Comment:
Test does not check if selected minOSVersion is being sent correctly via the network.